mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -48,8 +48,11 @@ private[util] sealed trait BaseReadWrite {
   /**
    * Returns the user-specified SQL context or the default.
    */
-  protected final def sqlContext: SQLContext = optionSQLContext.getOrElse {
-    SQLContext.getOrCreate(SparkContext.getOrCreate())
```

Comment:
Seems we still need to use getOrCreate? Otherwise, we will keep creating new sqlContexts. Also, if we actually use HiveContext, without using getOrCreate, we will not be able to get that.